server/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSourceConfig.java

```diff
@@ -61,6 +65,50 @@ public static ValuesSourceConfig resolve(QueryShardContext context,
                                              String format,
                                              ValuesSourceType defaultValueSourceType,
                                              String aggregationName) {
+
+        return internalResolve(context, userValueTypeHint, field, script, missing, timeZone, format, defaultValueSourceType,
+            aggregationName, ValuesSourceConfig::getMappingFromRegistry);
+    }
+
+    /**
+     * AKA legacy resolve.  This method should be called by aggregations not supported by the {@link ValuesSourceRegistry}, to use the
+     * pre-registry logic to decide on the {@link ValuesSourceType}.  New aggregations which extend from
+     * {@link ValuesSourceAggregationBuilder} should not use this method, preferring {@link ValuesSourceConfig#resolve} instead.
+     *
+     * @param context - the query context
+     * @param userValueTypeHint - User specified value type; used for missing values and scripts
+     * @param field - The field being aggregated over.  At least one of field and script must not be null
+     * @param script - The script the user specified.  At least one of field and script must not be null
+     * @param missing - A user specified value to apply when the field is missing.  Should be of type userValueTypeHint
+     * @param timeZone - Used to generate a format for dates
+     * @param format - The format string to apply to this field.  Confusingly, this is used for input parsing as well as output formatting
+     *               See https://github.com/elastic/elasticsearch/issues/47469
+     * @param defaultValueSourceType - per-aggregation {@link ValuesSource} of last resort.
+     * @return - An initialized {@link ValuesSourceConfig} that will yield the appropriate {@link ValuesSourceType}
+     */
+    public static ValuesSourceConfig resolveUnregistered(QueryShardContext context,
+                                                         ValueType userValueTypeHint,
+                                                         String field,
+                                                         Script script,
+                                                         Object missing,
+                                                         ZoneId timeZone,
+                                                         String format,
+                                                         ValuesSourceType defaultValueSourceType) {
+        return internalResolve(context, userValueTypeHint, field, script, missing, timeZone, format, defaultValueSourceType, null,
+            ValuesSourceConfig::getLegacyMapping);
+    }
+
+    public static ValuesSourceConfig internalResolve(QueryShardContext context,
```

Comment:
Should this be private, since it uses a private interface (`FieldResolver`) and looks to be an internal thing anyway?

Reply:
yes, yes it should.  Will fix before merging.  Thanks!